_posts/2020-04-01-Robust Apps for the cloud.md

```diff
@@ -0,0 +1,253 @@
+
+- Eliminates file lock conflicts between deployment and runtime.
+- Ensures only full-deployed apps are running at any time.
+- Can be deployed to a production app (with restart).
+- Improves the performance of Azure Resource Manager deployments.
+- May reduce cold-start times, particularly for JavaScript functions with large npm package trees.
+
+> Please note that this feature is not compatible with [local cache](#local-cache). Also if you are using a CMS application, we do **not** recommend the use of this feature.
+
+Refer to the documentation to [get started with Run from Package](https://docs.microsoft.com/azure/app-service/deploy-run-package). Ensure to set the app setting `WEBSITE_DISABLE_STANDBY_VOLUMES = 1`, this prevents the app from getting restarted when the primary storage volume is down and the app starts using standby storage volume. 
+
+> Please note the zip file should not exceed 1 GB.
+
+## Enable Local Cache
+
+When this feature is enabled, the site content is read, written from the local virtual machine instance instead of fetching from Azure storage (where site content is stored). This will reduce the number of recycles required for the app. It can be enabled through Azure portal from the "General -> Application settings". On this page under the App settings section add `WEBSITE_LOCAL_CACHE_OPTION` as key and `"Always"` as value. Also add the `WEBSITE_LOCAL_CACHE_SIZEINMB` with a desired local cache size value up to 2000MB (if not provided, it defaults to 300 MB). It helps to provide the cache size specially when the site contents are more than 300 MB. Ensure that site contents are less than 2000MB for this feature to work. Also it is a good practice to keep it as a slot setting so that it does not get removed with a swap.
+**The most important thing to keep in mind here** is that app should not be doing local disk writes for state persistence of its data/transactions.
+External storage like storage containers, db or cosmosDB should be used for storage purposes.
+
+> Please note that the behavior of Local Cache depends on the language and CMS you are using. For best results, we recommend using it for .net and .netcore apps as long as local writes are not being done by the app.  
+
+![multiple-instances]({{site.baseurl}}/media/2020/04/local-cache.png)
+
+![multiple-instances]({{site.baseurl}}/media/2020/04/local-cache-2.png)
+
+**Learn More**
+
+- [Azure App Service Local Cache overview](https://docs.microsoft.com/en-us/azure/app-service/overview-local-cache)
+
+## Auto Heal
+
+Enable auto heal on the app. I would recommend to create an auto heal mitigation rule by going to "Diagnose and Solve problems" section, selecting the "Diagnostic Tools" tile and then "Auto Healing" under Proactive Tools section.
```

Comment:
This paragraph should explain what autoheal is and what it does.

Reply:
Agreed. Added an reworded some mistakes.